tests/testapp/tests/test_connection.py

```diff
@@ -103,6 +114,24 @@ def test_csr(self):
             self.network_connection.certificate_signing_request(self.root_cert, '', '')
         self.assertTrue(Certificate.objects.filter(id=json.loads(cert_serialized)['id']).exists())
 
+    @mock_patch_decorator
+    def test_push_signed_client_certificate_chain(self):
+        # sanity test to make sure things are working as expected
+        NetworkSyncConnection._request.return_value.json.return_value = {'id': uuid.uuid4().hex, 'public_key': Key().get_public_key_string()}
+        cert = self.network_connection.push_signed_client_certificate_chain(self.root_cert,
+                                                                            self.subset_scope_def.id,
+                                                                            {"mainpartition": self.root_cert.id, "subpartition": "abracadabra"})
+        self.assertEqual(cert.private_key, None)
+
+    @mock_patch_decorator
+    def test_push_signed_client_certificate_chain_403(self):
+        # access denied when requesting public key endpoint
+        NetworkSyncConnection._request.side_effect = HTTPError(response=mock.Mock(status_code=403))
+        with self.assertRaises(MorangoServerDoesNotAllowNewCertPush):
+            self.network_connection.push_signed_client_certificate_chain(self.root_cert,
+                                                                         self.subset_scope_def.id,
+                                                                         {"mainpartition": self.root_cert.id, "subpartition": "abracadabra"})
+
```

Comment:
General thought about the tests: You seem to be separately testing either the API endpoints or the code that calls them. Would it be possible to do a few "integration tests" of calling something like `push_signed_client_certificate_chain` and having it hit the endpoints, and make sure they play well together?